### PR TITLE
Typing library to skip bad dates (configurable)

### PR DESCRIPTION
**Pull request by Tang8330**
> [!NOTE]
> Add SkipBadTimestamps setting to return null for unsupported date/timestamp values during staging, using new typed ParseError to detect unsupported date layouts.
> 
> - **Snowflake Staging**:
>   - `castColValStaging` now accepts `config` and, when `sharedDestinationSettings.skipBadTimestamps` is true, returns `__artie_null_value` instead of error for bad timestamps detected via typed parse errors.
> - **Config**:
>   - Add `sharedDestinationSettings.skipBadTimestamps` flag.
> - **Typing**:
>   - Introduce `ParseError` with `UnsupportedDateLayout` kind and helpers; `ParseDateFromAny` now returns `ParseError` for unsupported layouts.
> - **Tests**:
>   - Add tests verifying null-return vs error behavior based on `SkipBadTimestamps`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4bc94c8f03868c65fbcbd7891efbf6753ce507e1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by williamhaw — APPROVED** [review]
lgtm % c

File: clients/snowflake/staging.go [hunk 1]
```diff
@@ -37,13 +37,21 @@ func replaceExceededValues(colVal string, kindDetails typing.KindDetails) string
...
-func castColValStaging(colVal any, colKind typing.KindDetails, _ config.SharedDestinationSettings) (shared.ValueConvertResponse, error) {
+func castColValStaging(colVal any, colKind typing.KindDetails, config config.SharedDestinationSettings) (shared.ValueConvertResponse, error) {
 	if colVal == nil {
 		return shared.ValueConvertResponse{Value: constants.NullValuePlaceholder}, nil
 	}
 
 	value, err := values.ToString(colVal, colKind)
 	if err != nil {
+		if config.SkipBadTimestamps {
+			if parseError, ok := typing.BuildParseError(err); ok {
```
[the remark this answers is not in the document]
Comment (Tang8330) [reply clients/snowflake/staging.go:48]: yeah why not
